sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ClientSideRequestStatistics.java

```diff
@@ -203,77 +217,72 @@ void setRegionsContacted(Set<URI> regionsContacted) {
         this.regionsContacted = Collections.synchronizedSet(regionsContacted);
     }
 
-    private static String formatDateTime(ZonedDateTime dateTime) {
-        if (dateTime == null) {
-            return null;
-        }
-        return dateTime.format(RESPONSE_TIME_FORMATTER);
-    }
-
-    public void recordRetryContext(RxDocumentServiceRequest request) {
-        if(request.requestContext.retryContext != null) {
-            request.requestContext.retryContext.retryEndTime =  ZonedDateTime.now(ZoneOffset.UTC);
+    void recordRetryContext(RxDocumentServiceRequest request) {
+        if (request.requestContext.retryContext != null) {
+            request.requestContext.retryContext.retryEndTime = ZonedDateTime.now(ZoneOffset.UTC);
             this.retryContext = new RetryContext(request.requestContext.retryContext);
         }
     }
 
     static class StoreResponseStatistics {
         @JsonSerialize(using = StoreResult.StoreResultSerializer.class)
-        public StoreResult storeResult;
+        StoreResult storeResult;
         @JsonSerialize(using = ZonedDateTimeSerializer.class)
-        public ZonedDateTime requestResponseTime;
-        public ResourceType requestResourceType;
-        public OperationType requestOperationType;
+        ZonedDateTime requestResponseTime;
```

Comment:
Please verify that after moving these field(which doesn't have a explicit serializer)from public to default to after are still coming in diagnostic ,I think  we need to do it manually as all field level validation is not their in tests , please refer CosmosResponseDIagnoticsTest file

Reply:
very good point. also as this PR adds getters to the classes in ClientSideRequestStatistics, jackson may see to ways to get a feild, the field itself and the getter for the field.

We should ensure they are properly annotated as per jackson guideline if one need to be ignored and the other one be used. etc.

Reply:
validated with @simplynaveen20 . These look good